### PR TITLE
Implement contrast for settings page

### LINKED ISSUE #62: Implement the actual settings on the front-end side
We're saving the user's data in the database, but we need to actually make it work now.

### LINKED ISSUE #91: Implement Contrast
Currently the user can specify if they want a high contrast or low contrast.

The application should reflect those changes.

### PR DESCRIPTION
**Pull request by sderacy**
Closes #62, Closes #91

I got this working on the settings page before #135 but I figured I'm gonna have to change this up a bit. I'll let you know when it's ready.

### CODE REVIEW

File: views/account/settings/main.js
Comment (paytonshaltis) [inline views/account/settings/main.js:146]: I know you're still working on this, but I want to let you know early about this code. From the theme flashing bug fix, none of this code is in this file anymore so you'll probably have conflicts.

Comment (sderacy) [reply views/account/settings/main.js:146]: Yup! I'm currently refactoring it since I saw that you had moved it out of the file.

Shouldn't be too bad though.

Thank you for the heads up!

Comment (paytonshaltis) [reply views/account/settings/main.js:146]: No problem chief, figured I'd try and prevent a headache later!

**Comment by sderacy** [comment]
This is most of the code for the contrast.

Known bugs: 
-When you click to high contrast and try to switch between themes, until you save it, it will show the wrong contrast theme color.
-The home page labels does not cast the contrast box on just the labels because of a weird issue with javascript. I haven't done too much reading on this, but I think there is a race condition happening between the two js files and essentially, the classes I apply to the elements created in home/main.js are not visible in time for when we apply the contrast. @lelekaz what do you think?

I made the contrast trying to keep in mind what @thadakn1 and I talked about over text, so let me know what you guys think
@paytonshaltis @matyis1 @lelekaz @thadakn1 

This is probably my last commit for the night.

I apologize in advance for the color overload that'll happen to your eyes 😅

**Comment by sderacy** [comment]
Also for the home page, the contrast would be applied to just the text, so this is not what it should look like permanently at all.

**Comment by thadakn1** [comment]
great work so far sterly! it's looking so cool!! i made some changes on my end locally, I limited the background-color for h1s and h2s to just the content so it doesn't stretch across the page. I also fixed that radio button spacing and added some padding to some h1 and h2 tags for readability reasons. i changed the "Edit Existing" and "Add a new account" text to h2s from h1s because logically speaking it is a sub-topic or sub-section of the entire Profile section which is a <h1> tag itself.

also... i dont know what i messed with that on the home page the toggles and the labels all had background applicable on them. maybe i didn't see them earlier, but they randomly appeared like that (unless i touched something i didn't mean to). 

just wanted to ask if you if I should push my local changes to this Pull Request so you can see them?  @sderacy I won't do anything until you give a green light so just lmk. in the mean time, i'm going to be working on another branch maybe with the help feature and some other issues I claimed

**Comment by sderacy** [comment]
> also... i dont know what i messed with that on the home page the toggles and the labels all had background applicable on them. maybe i didn't see them earlier, but they randomly appeared like that (unless i touched something i didn't mean to).

If I understand what you're referring to, no worries. You did not cause this bug. I mentioned this in the earlier comment:

>The home page labels does not cast the contrast box on just the labels because of a weird issue with javascript. I haven't done too much reading on this, but I think there is a race condition happening between the two js files and essentially, the classes I apply to the elements created in home/main.js are not visible in time for when we apply the contrast. @lelekaz what do you think?

Essentially, we're creating some elements using javascript and my current approach for implementing contrast is assigning a "contrast" class to each text element that you can then manipulate in the defaultSettings.js file. The problem is that with the elements created via javascript, the contrast class isn't being modified once it's been applied because I think the creation/retrieval time of the element is happening after the defaultSettings file is executed.
 
> just wanted to ask if you if I should push my local changes to this Pull Request so you can see them? @sderacy I won't do anything until you give a green light so just lmk. in the mean time, i'm going to be working on another branch maybe with the help feature and some other issues I claimed

If you want to push your changes up, what exactly did you want to add? I'm not generally opposed to you working on the branch but I wanna know what you plan on changing since it'll give me a better idea as i'm probably gonna get back to working on this sometime this weekend.

**Comment by sderacy** [comment]
> great work so far sterly! it's looking so cool!! i made some changes on my end locally, I limited the background-color for h1s and h2s to just the content so it doesn't stretch across the page. I also fixed that radio button spacing and added some padding to some h1 and h2 tags for readability reasons. i changed the "Edit Existing" and "Add a new account" text to h2s from h1s because logically speaking it is a sub-topic or sub-section of the entire Profile section which is a h1 tag itself.

Nevermind, just saw this. Yea you can push this up.

Just to clarify though, by 

> I also fixed that radio button spacing

Do you mean on the settings page? If so, we can change that in this PR but I think it would make more sense to restructure the settings page in general.

**Comment by paytonshaltis** [comment]
> When you click to high contrast, save it, and try to switch between themes, until you save changes after switching themes, it will show the wrong contrast theme color. @paytonshaltis @matyis1 I may have to work with you guys to figure that out.

Fixed. The issue is that the function that updates the display on form input change was not updated with the contrast code; it is working now.

**Comment by paytonshaltis** [comment]
> The home page labels does not cast the contrast box on just the labels because of a weird issue with javascript. I haven't done too much reading on this, but I think there is a race condition happening between the two js files and essentially, the classes I apply to the elements created in home/main.js are not visible in time for when we apply the contrast.

Also fixed. I added a div on the page with display: none and class of contrast. This way, when the labels are being generated, the class that is on this element (either high contrast dark, high contrast light, or none) can also be added to the labels when they are being created.

**Comment by sderacy** [comment]
Nice job @paytonshaltis!

That looks a lot nicer haha

@thadakn1 I think for the most part, I'm done making changes. If you want, I'd recommend pulling the code down and pushing your code up.

**Comment by sderacy** [comment]
There is one more element that doesnt have the contrast being applied to it:
image

It's a similar issue to the toggle labels. @paytonshaltis could what you did be applied in the same manner for this innerHTML that was used to set this up?

**Comment by paytonshaltis** [comment]
> It's a similar issue to the toggle labels. @paytonshaltis could what you did be applied in the same manner for this innerHTML that was used to set this up?

On it partner

**Comment by sderacy** [comment]
Looks great! I think I found one last bug, which happens when you select a theme, select high contrast, save changes, and then select another theme. 

It gets fixed when you click save changes but figured I should report this in case we think we wanna fix it.

image

Other than that, looks perfect.

**Comment by paytonshaltis** [comment]
> Looks great! I think I found one last bug, which happens when you select a theme, select high contrast, save changes, and then select another theme.

All good now. The issue is that switching themes applied both the `high-contrast-light` and `high-contrast-dark` classes to these elements, and the light theme contrast CSS properties apparently have higher precedence. The solution is to always just remove the other class when adding one of these classes.

**Comment by sderacy** [comment]
Looks good! Seriously good job with all of this @paytonshaltis and thank you for taking a look @thadakn1. Let me know what other thoughts you have on this PR.

If everything looks good, just approve this and we can merge this in.

**Review by paytonshaltis — APPROVED** [review]
LGTM. I'll approve, but if anyone else has anything to mention before the merge feel free.

**Pull request merged**
